src/middlewares/request-unflattener.js

```diff
@@ -0,0 +1,79 @@
+import _ from 'lodash';
+
+function detectFlattened(name) {
+  return name?.includes('|');
+}
+
+function getParentFieldName(fieldName) {
+  return fieldName?.split('|')[0];
+}
+
+function unflattenCollectionFields(requestedFields) {
+  const fieldNames = new Set();
+  requestedFields.split(',')
+    .forEach((requestedField) => fieldNames.add(getParentFieldName(requestedField)));
+  return [...fieldNames].join(',');
+}
+
+function unflattenFields(request) {
+  Object.entries(request.query.fields).forEach(([collection, requestedFields]) => {
+    if (detectFlattened(requestedFields)) {
+      request.query.fields[collection] = unflattenCollectionFields(requestedFields);
+    }
+  });
+}
+
+function unflattenAttribute(attributeName, value, attributes) {
+  let accessPathArray = attributeName.split('|');
+  const parentObjectName = accessPathArray.shift();
+  accessPathArray = accessPathArray.reverse();
+  const parentObject = attributes[parentObjectName] || {};
+  const unflattenedProperty = accessPathArray.reduce((a, prop) => ({ [prop]: a }), value);
+  const unflattenedObject = _.merge(parentObject, unflattenedProperty);
```

Comment:
[FINE TUNING][PERFORMANCE] I'm wondering if this can't be approved, the `merge` seems like a little bit overkill.

Reply:
Too hard to do something else